### DISCUSSION TITLE
[Android] Unlock script hash is incorrect after btc signature

### DISCUSSION THREAD

**Discussion by YaakovHuang** [trustwallet/wallet-core#1375] [Q&A]
**Describe the bug**
Unlock script hash is incorrect after bitcoin signature

I am going to send a transaction from the P2PKH address to the Bech32 address, but the hash of the unlocking script is incorrect.
> from 1HQ3rb7nyLPrjnuW85MUknPekwkn7poAUm to bc1qceyq9k3uer7m2qa6zf9z3rxjzxa95k5pknmjja

```
HDWallet wallet = new HDWallet("army van defense carry jealous true garbage claim echo media make crunch", "");
CoinType bitcoin = CoinType.BITCOIN;
PrivateKey privateKey = wallet.getKey(CoinType.BITCOIN, "m/44'/0'/0'/0/0");
ECKey ecKey = ECKey.fromPrivate(privateKey.data());
PublicKey publicKey = privateKey.getPublicKeySecp256k1(true);
BitcoinAddress address = new BitcoinAddress(publicKey, (byte) 0x00);
Logger.d("pk = " + ecKey.getPrivateKeyAsWiF(MainNetParams.get()));
Logger.d("publicKey = " + publicKey.description());
Logger.d("address = " + address.description());
// 1HQ3rb7nyLPrjnuW85MUknPekwkn7poAUm
String fromAddress = address.description();
String toAddress = "bc1qceyq9k3uer7m2qa6zf9z3rxjzxa95k5pknmjja";

BitcoinScript script = BitcoinScript.lockScriptForAddress(fromAddress, bitcoin);
byte[] prevHash = NumericUtil.reverseBytes(ExtensionsKt.hexStringToByteArray("5932892f8dba160ed3608130083e4fc6f747b5113c12f4a2e7cf3376ca3291c9"));
Bitcoin.OutPoint outPoint = Bitcoin.OutPoint.newBuilder()
        .setHash(ByteString.copyFrom(prevHash))
        .setIndex(0)
        .setSequence((int) Long.MAX_VALUE)
        .build();
Bitcoin.UnspentTransaction utxo = Bitcoin.UnspentTransaction.newBuilder()
        .setAmount(1000)
        .setOutPoint(outPoint)
        .setScript(ByteString.copyFrom(script.data()))
        .build();
Bitcoin.SigningInput.Builder input = Bitcoin.SigningInput.newBuilder()
        .setAmount(750)
        .setHashType(BitcoinSigHashType.ALL.value())
        .setToAddress(toAddress)
        .setChangeAddress(fromAddress)
        .setByteFee(1)
        .setCoinType(bitcoin.value())
        .setUseMaxAmount(true)
        .addUtxo(utxo)
        .addPrivateKey(ByteString.copyFrom(privateKey.data()));

Bitcoin.TransactionPlan plan = AnySigner.plan(input.build(), bitcoin, Bitcoin.TransactionPlan.parser());
Logger.d("plan fee = " + plan.getFee() + ", amount = " + plan.getAmount() + ", avail_amount = " + plan.getAvailableAmount() + ", change = " + plan.getChange());
input.setPlan(plan);
input.setAmount(plan.getAmount());

Bitcoin.SigningOutput output = AnySigner.sign(input.build(), bitcoin, Bitcoin.SigningOutput.parser());
if (output.getError() != null) {
    Logger.e("output error = " + output.getError().toString());
}
byte[] signedTransaction = output.getEncoded().toByteArray();
Logger.d("signed transaction = " + Numeric.toHexStringNoPrefix(signedTransaction));
```
the binary data:
0100000001c99132ca7633cfe7a2f4123c11b547f7c64f3e08308160d30e16ba8d2f893259000000006a47304402206c4b600c7251a8ab26e13ad6c9d4a0ba8e479582946f37c990e986ca57d8877c0220363778d7487a7b06a37fabe8d8b3f8a79babcbd57bd65d09c83d788037f7e1bd0121036740c4f55d64fb6c9bc412084638b80062cee07f6c84b205671584e82a7c96b7ffffffff012b03000000000000160014c64802da3cc8fdb503ba124a288cd211ba5a5a8100000000
<img width="1167" alt="WX20210415-112220@2x" src="https://user-images.githubusercontent.com/9527803/114809553-fa249700-9ddc-11eb-9027-81df8faf3161.png">

the correct result:
https://api.blockcypher.com/v1/btc/main/txs/7a0946bfc5ea4872c3018c0fd24033cd4e4ad02a3753c5c7ae9a79ce1e6689f8
<img width="1167" alt="WX20210415-112220@2x" src="https://user-images.githubusercontent.com/9527803/114809658-2b9d6280-9ddd-11eb-9224-10a850a5320b.png">

**Comment by optout21** [comment]
Is this the same as https://github.com/trustwallet/wallet-core/issues/1365 ?

**Reply by YaakovHuang** [reply]
> Is this the same as #1365 ?

Hi, this problem is not the same problem as #1365 problem. I have solved the #1365 problem. The problem encountered now is signed by me. The problem now is unlock script hash is incorrect, and ios also encountered the same problem.

**Reply by edenmeshulam** [reply]
Hi, 
Someone solved this in android ?
